pkg/controller/container-runtime-config/helpers.go

```diff
@@ -933,17 +934,74 @@ func validateClusterImagePolicyWithAllowedBlockedRegistries(clusterScopePolicies
 	return nil
 }
 
-func generateSigstoreRegistriesdConfig(clusterScopePolicies map[string]signature.PolicyRequirements) ([]byte, error) {
+func generateSigstoreRegistriesdConfig(clusterScopePolicies map[string]signature.PolicyRequirements, registriesTOML []byte) ([]byte, error) {
 	if len(clusterScopePolicies) == 0 {
 		return nil, nil
 	}
 
-	registriesDockerConfig := make(map[string]dockerConfig)
-	sigstoreAttachment := dockerConfig{
-		UseSigstoreAttachments: true,
+	var (
+		err                    error
+		tmpFile                *os.File
+		registriesConf         = sysregistriesv2.V2RegistriesConf{}
+		registriesDockerConfig = make(map[string]dockerConfig)
+		sigstoreAttachment     = dockerConfig{
+			UseSigstoreAttachments: true,
+		}
+		sysContext = &types.SystemContext{
```

Comment:
(This is only used on the `registriesTOML != nil` path, so it can be initialized there. A context with `SystemRegistriesConfPath` set but `SystemRegistriesConfPath` does not make all that much sense. But this works fine as is.)